api/server/sdk/volume_ops.go

```diff
@@ -0,0 +1,226 @@
+/*
+Package sdk is the gRPC implementation of the SDK gRPC server
+Copyright 2018 Portworx
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package sdk
+
+import (
+	"context"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/libopenstorage/openstorage/api"
+	"github.com/libopenstorage/openstorage/pkg/util"
+	"github.com/portworx/kvdb"
+)
+
+// Create creates a volume
+func (s *VolumeServer) Create(
+	ctx context.Context,
+	req *api.VolumeCreateRequest,
+) (*api.VolumeCreateResponse, error) {
+
+	if req.GetLocator() == nil {
+		return nil, status.Error(
+			codes.InvalidArgument,
+			"Must supply a locator object")
+	} else if len(req.GetLocator().GetName()) == 0 {
+		return nil, status.Error(
+			codes.InvalidArgument,
+			"Must supply a unique name in locator")
+	} else if req.GetSource() == nil {
+		return nil, status.Error(
+			codes.InvalidArgument,
+			"Must supply source object")
+	} else if req.GetSpec() == nil {
+		return nil, status.Error(
+			codes.InvalidArgument,
+			"Must supply spec object")
+	}
+
+	locator := req.GetLocator()
+	spec := req.GetSpec()
+	source := req.GetSource()
+	volName := locator.GetName()
+
+	// Check if the volume has already been created or is in process of creation
+	v, err := util.VolumeFromName(s.driver, volName)
+	if err == nil {
+		// Check the requested arguments match that of the existing volume
+		if spec.Size != v.GetSpec().GetSize() {
+			return nil, status.Errorf(
+				codes.AlreadyExists,
+				"Existing volume has a size of %v which differs from requested size of %v",
+				v.GetSpec().GetSize(),
+				spec.Size)
+		}
+		if v.GetSpec().GetShared() != req.GetSpec().GetShared() {
```

Comment:
You might just want to add a new function that compares two specs and then returns an error if there is a discrepancy. It will compare all the fields and also handle empty values.